src/compiler/crystal/command.cr

```diff
@@ -447,7 +447,19 @@ class Crystal::Command
 
       unless no_codegen
         opts.on("--release", "Compile in release mode") do
-          compiler.release = true
+          compiler.optimization_mode = Compiler::OptimizationMode::Level3
+        end
+        opts.on("--level3", "Compile in level3 (--release) mode") do
+          compiler.optimization_mode = Compiler::OptimizationMode::Level3
+        end
+        opts.on("--level2", "Compile in level2 mode") do
+          compiler.optimization_mode = Compiler::OptimizationMode::Level2
+        end
+        opts.on("--level1", "Compile in level1 mode") do
+          compiler.optimization_mode = Compiler::OptimizationMode::Level1
+        end
+        opts.on("--level0", "Compile in level1 mode") do
+          compiler.optimization_mode = Compiler::OptimizationMode::Level0
```

Comment:
What about more following the optimization levels of compilers like [GCC](https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html) and [LLVM](https://clang.llvm.org/docs/CommandGuide/clang.html#code-generation-options)?

Instead of "--level2" it would be "-O2"?